### DISCUSSION TITLE
Span Status shows OK for 1xx,  2xx, 3xx HTTP status code

### DISCUSSION THREAD

**Discussion by debasis9dash** [open-telemetry/opentelemetry-specification#2596] [Q&A]
Hello,

According to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#Status span status should be kept as Unset for the 1xx, 2xx and 3xx HTTP status codes. 
The description doesn't tell on what scenario span status should be OK.
Should the Span status be OK when the HTTP status code are in 1xx ,2xx, 3xx?

**Comment by Oberon00** [comment]
IIRC, the OK span status should never be set by instrumentations, it should only be used as an explicit override by applications. Consult the API specification for SetStatus: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status

> `Ok`:  The operation has been validated by an Application developer or Operator to have completed successfully.

and

> Generally, Instrumentation Libraries SHOULD NOT set the status code to Ok, unless explicitly configured to do so.
